git ignore compiled python files, log files, cache, and saves

jag426 opened the pull request:
I used renpy's own .gitignore file for advice.
To remove previously checked in files matching the new .gitignore rules, I ran something like 'find . -type f -name "*.rpyc" | xargs git rm' for each new wildcard rule.

Testing:
I ran 'git clean -f'. Then, I tested launching and playing the game, building the game, and extracting and playing the built game. I also tested saving and loading. (Curiously, old saved games still show up in save slots, even though I deleted game/saves/.) This verifies that these files do not need to be checked in.
I also checked that all these files got regenerated by launching the game, and that despite this, 'git status' no longer lists them.

develop4learners commented:
I checked out your branch. git clean -f did not get rid of all untracked files for me. Before, I had not checked in, many saves, the game/cache/bytecode.rpyb, the persistent, and the log.txt. After git clean -f these files remained. Instead, got rid of untracked files using git checkout -- *. Hard to find the gitignore but found it using find ./ -name ".gitignore"
PLAY: yes! Git ignored my saves, log, etc. changes. git status returned no changes after Launch Project and after pre-Hack1 play.
SAVE: yes, same as above
ERRORS: Game threw a StoreModule error, but git status didn't know about it. Nice. 
LOAD: yes, loaded a post-Hack1 save and git status had no changes. Nice. 

ISSUE: I am not sure how you got errors.txt to regenerate. I cannot get that file to show up again.
OTHER CHANGES TO IGNORE? After looking for the "errors.txt" file.    git started to notice   .DS_Store had changes not staged for commit.

jag426 commented:
> I checked out your branch. git clean -f did not get rid of all untracked files for me.
The proper command is actually "git clean -Xf" for removing ignored files. Fixed.

> OTHER CHANGES TO IGNORE? After looking for the "errors.txt" file. git started to notice .DS_Store had changes not staged for commit.
Added .DS_Store to the .gitignore and removed it from version control as well.
